cassandra-scheduler/src/main/java/io/mesosphere/mesos/frameworks/cassandra/PersistedCassandraClusterState.java

```diff
@@ -78,8 +86,35 @@ public void executorMetadata(@NotNull final List<CassandraFrameworkProtos.Execut
      * Add a node, making sure to replace any previoud node with the same hostname
      */
     public void addOrSetNode(final CassandraFrameworkProtos.CassandraNode node) {
-        final FluentIterable<CassandraFrameworkProtos.CassandraNode> filterNot = from(nodes())
-            .filter(not(cassandraNodeHostnameEq(node.getHostname())));
-        nodes(append(newArrayList(filterNot), node));
+        List<CassandraFrameworkProtos.CassandraNode> nodeList = new ArrayList<>(nodes());
+        for (int i = 0; i < nodeList.size(); i++) {
+            CassandraFrameworkProtos.CassandraNode candidate = nodeList.get(i);
+            if (node.getHostname().equals(candidate.getHostname())) {
+                nodeList.set(i, node);
+                nodes(nodeList);
+                return;
+            }
+        }
+        nodeList.add(node);
+        nodes(nodeList);
+    }
+
+    public Tuple2<Integer, Integer> nodeCounts() {
```

Comment:
Given that both items are Integer it's will be confusing trying to remember which tuple value is seeds and which is nodes. Create a more expressive type to be the return type for this method.